corehq/apps/reports/views.py

```diff
@@ -1511,7 +1511,7 @@ def edit_case_view(request, domain, case_id):
     if update:
         submit_case_blocks([CaseBlock(
             case_id=case_id,
-            owner_id=case.owner_id,
+            external_id=case.external_id,
```

Comment:
this seems like a strange way to pass this value through. I would have just used a variable, instead of storing it temporarily on the case object. It will also not update if there is only a change to `external_id` (and not other case properties, since `update` will be empty. In addition, this will also look as though the external ID changes with every edit case, even if it doesn't. You may want to do something like:

```py
case_kwargs = {}
if update: 
    case_kwargs['update'] = update
if 'external_id' in request.POST and request.POST['external_id'] != case.external_id:
    case_kwargs['external_id'] = request.POST['external_id']
...
if case_kwargs:
    submit(CaseBlock(**case_kwargs))
```

Reply:
https://github.com/dimagi/commcare-hq/pull/19071/commits/f74e7d0a7deef7ba7b1ab183567bf4b69d5e94e0